### PR TITLE
Feature/hht 1046 link dataset to model

### PR DESCRIPTION
**Pull request by Kagameow**

### CODE REVIEW

File: src/app/components/model-dataset-link/model-dataset-link.component.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+  ];
+
+  constructor(
+    private cogFrameworkApiService: CogFrameworkApiService,
+    private snackBar: MatSnackBar,
+    private translocoService: TranslocoService,
+  ) {}
+
+  private openSnackBar(message: string, action: string): void {
+    this.snackBar.open(message, action, {
```
Comment (kaaboeld) [inline src/app/components/model-dataset-link/model-dataset-link.component.ts]: Maybe move this to consts?

File: src/app/pages/model-detail/model-datasets/model-datasets.component.html [hunk 2]
```diff
@@ -11,26 +16,38 @@
             [dataSource]="modelDatasetDataSource"
```
Comment (kaaboeld) [inline src/app/pages/model-detail/model-datasets/model-datasets.component.html]: Rename:
`modelDatasetDataSource > dataSource`

Redata:
`modelDatasetDataSource: ModelDatasetInfo[] = []; > dataSource = new MatTableDataSource<DatasetData>([]);`

File: src/app/pages/model-detail/model-datasets/model-datasets.component.ts [hunk 3]
```diff
@@ -19,10 +24,61 @@ export class ModelDatasetsComponent {
 
   _modelDetailData!: ModelDetailData;
   modelDatasetDataSource: ModelDatasetInfo[] = [];
```
Comment (kaaboeld) [inline src/app/pages/model-detail/model-datasets/model-datasets.component.ts]: Up

File: src/app/pages/model-detail/model-datasets/model-datasets.component.ts [hunk 4]
```diff
@@ -19,10 +24,61 @@ export class ModelDatasetsComponent {
...
+  ];
+
+  constructor(
+    private cogFrameworkApiService: CogFrameworkApiService,
+    private snackBar: MatSnackBar,
+    private translocoService: TranslocoService,
+  ) {}
+
+  private openSnackBar(message: string, action: string): void {
+    this.snackBar.open(message, action, {
```
Comment (kaaboeld) [inline src/app/pages/model-detail/model-datasets/model-datasets.component.ts]: Up

**Review by kaaboeld — CHANGES_REQUESTED** [review]

**Comment by Kagameow** [comment]
I'll merge since I'm a little blocked. Please, Eugene, let me know if there are any other issues regarding the pr, I will fix them in a separate pr.

**Pull request merged**